lib/Reply.js

```diff
@@ -141,10 +143,26 @@ class Reply {
 }
 
 
-function cannonicalizeDirectives(directives) {
+function cannonicalizeDirectives(alexaEvent, directives) {
   if (!directives) return [];
   if (!_.isArray(directives)) directives = [directives];
-  return directives.map(cannonicalizeDirective);
+
+  if (_.filter(directives, { type: 'Display.RenderTemplate' }).length > 1) {
+    throw new Error('At most one Display.RenderTemplate directive can be specified in a response');
+  }
+
+  if (_.find(directives, { type: 'AudioPlayer.Play' }) && _.find(directives, { type: 'VideoApp.Launch' })) {
+    throw new Error('Do not include both an AudioPlayer.Play directive and a VideoApp.Launch directive in the same response');
+  }
+
+  return directives
+  .filter((directive) => {
+    const hasDisplayInterface = _.get(alexaEvent, 'context.System.device.supportedInterfaces.Display');
+    const directivesType = directive.type;
+    if (!hasDisplayInterface && _.includes(['Display.RenderTemplate', 'Hint'], directivesType)) return false;
```

Comment:
This is key piece of functionality. Can we pull it out into a separate function to make it clearer?

Reply:
Done!